openhcl/hcl/src/ioctl.rs

```diff
@@ -3205,4 +3251,36 @@ impl Hcl {
             hcl_tlbsync(self.mshv_vtl.file.as_raw_fd()).expect("should always succeed");
         }
     }
+
+    /// Get vtl0 target vector for remapped device interrupt
+    pub fn get_device_irr_map(&self, target_vector: u8) -> Option<u8> {
+        let remaped_vector = self.device_irr_target_to_remapped_vector[target_vector as usize];
+        if remaped_vector == 0 {
+            return None;
+        }
+        Some(remaped_vector)
+    }
+
+    /// Update vtl0 device interrupt map
+    pub fn set_device_irr_map(&self, remapped_vector: u8, target_vector: u8) {
+        // update target(vtl0) to remaped(vtl2) vector mapping
+        let device_irr_remapped_vector = unsafe { &*(addr_of!(self.device_irr_target_to_remapped_vector[target_vector as usize]).cast::<AtomicU8>()) };
```

Comment:
this unsafe will not fly - either declare it as atomic u8 upfront or we may need a different kind of data structure if we do expect multiple vps to access this, which is possible

Reply:
Agreed. do you have any suggestions on where the vector mapping information can be stored? Currently I am storing it in the Hcl struct. 

We want to write to this structure from the intercept routine for retarget_interrupt hypercall but this does not have mutable access to Hcl structure.

Reply:
this is shared partition wide, right? let me think about it.

Reply:
Correct. It is partition wide. Update to the struct is needed only during device initialization but needs to be read during every device interrupt delivery.